src/internal/pubsub/publisher.hpp

```diff
@@ -0,0 +1,120 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2021-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "internal/pubsub/base.hpp"
+#include "internal/resources/forward.hpp"
+#include "internal/ucx/endpoint.hpp"
+
+#include "srf/codable/forward.hpp"
+#include "srf/node/source_channel.hpp"
+#include "srf/pubsub/api.hpp"
+#include "srf/runtime/remote_descriptor.hpp"
+#include "srf/utils/macros.hpp"
+
+#include <cstddef>
+#include <cstdint>
+#include <string>
+#include <unordered_map>
+#include <utility>
+
+namespace srf::internal::runtime {
+class Partition;
+}
+
+namespace srf::internal::pubsub {
+
+class Publisher : public Base,
```

Comment:
Can we call this `PublisherService` to distinguish it from the public `Publisher`?

Reply:
Done